File: client/public/Codes/C/SpecificKeyword.c
#include <stdio.h>
#include <string.h>
#define MAX_SIZE 100

int countOccurrences(char *str, char *toSearch);

int main()
{
    char str[MAX_SIZE];
    char toSearch[MAX_SIZE];
    int count;

    printf("Enter any string: ");
    gets(str);
    printf("Enter word to search occurrences: ");
    gets(toSearch);

    count = countOccurrences(str, toSearch);

    printf("%d", count);

    return 0;
}

int countOccurrences(char *str, char *toSearch)
{
    int i, j, found, count;
    int stringLen, searchLen;

    stringLen = strlen(str);
    searchLen = strlen(toSearch);

    count = 0;

    for (i = 0; i <= stringLen - searchLen; i++)
    {
        found = 1;
        for (j = 0; j < searchLen; j++)
        {
            if (str[i + j] != toSearch[j])
            {
                found = 0;
                break;
            }
        }

        if (found == 1)
        {
            count++;
        }
    }

    return count;
}

File: client/public/Codes/Java/SpecificKeyword.java
import java.util.Scanner;

public class SpecificKeyword {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String str;
        String toSearch;
        int count;
        System.out.print("Enter any string: ");
        str = scanner.nextLine();
        System.out.print("Enter word to search occurrences: ");
        toSearch = scanner.nextLine();
        count = countOccurrences(str, toSearch);
        System.out.println(count);

        scanner.close();
    }

    public static int countOccurrences(String str, String toSearch) {
        int i, j, found, count;
        int stringLen, searchLen;
        stringLen = str.length();
        searchLen = toSearch.length();
        count = 0;
        for (i = 0; i <= stringLen - searchLen; i++) {
            found = 1;
            for (j = 0; j < searchLen; j++) {
                if (str.charAt(i + j) != toSearch.charAt(j)) {
                    found = 0;
                    break;
                }
            }
            if (found == 1) {
                count++;
            }
        }
        return count;
    }
}
